### PR TITLE
Archived Delete API

### PR DESCRIPTION
**Pull request by melvl01**
Based on delete command, created delete from archive API. Added unit tests and system tests.

### CODE REVIEW

File: packages/workflows/src/api/ArchivedDelete.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+export class ArchivedDeleteWorkflow {
+
+    /**
+     * Delete a workflow from an archived
+     * @param {AbstractSession} session                     - z/OSMF connection info
+     * @param {string} workflowKey                          - Unique identifier of the workflow instance.
+     * @param {string} zOSMFVersion                         - Identifies the version of the zOSMF workflow service.
+     * @returns {string}
+     */
+ public static async archivedDeleteWorkflow(session: AbstractSession, workflowKey: string,
```
Comment (uzuko01) [inline packages/workflows/src/api/ArchivedDelete.ts]: can we have 4 spaces of indentation ? : )

File: packages/workflows/src/api/ArchivedDelete.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import { WorkflowConstants, nozOSMFVersion, noWorkflowKey } from "./WorkflowConstants";
+import { WorkflowValidator } from "./WorkflowValidator";
+
+/**
+ * Class to handle deletion of workflow from an archived z/OSMF workflow from a z/OS system.
+ */
+export class ArchivedDeleteWorkflow {
+
+    /**
+     * Delete a workflow from an archived
```
Comment (uzuko01) [inline packages/workflows/src/api/ArchivedDelete.ts]: Delete archived workflow ?

File: packages/workflows/src/api/ArchivedDelete.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+*/
+
+
+import { AbstractSession } from "@brightside/imperative";
+import { ZosmfRestClient } from "../../../rest";
+import { WorkflowConstants, nozOSMFVersion, noWorkflowKey } from "./WorkflowConstants";
+import { WorkflowValidator } from "./WorkflowValidator";
+
+/**
+ * Class to handle deletion of workflow from an archived z/OSMF workflow from a z/OS system.
```
Comment (uzuko01) [inline packages/workflows/src/api/ArchivedDelete.ts]: Class to  handle deletion of an archived z/OSMF workflow from a z/OS system?

File: packages/workflows/src/api/ArchivedDelete.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+     */
+ public static async archivedDeleteWorkflow(session: AbstractSession, workflowKey: string,
+                                            zOSMFVersion = WorkflowConstants.ZOSMF_VERSION){
+    WorkflowValidator.validateSession(session);
+    WorkflowValidator.validateNotEmptyString(zOSMFVersion, nozOSMFVersion.message);
+    WorkflowValidator.validateNotEmptyString(workflowKey, noWorkflowKey.message);
+    let resourcesQuery: string = `${WorkflowConstants.RESOURCE}/${zOSMFVersion}/`;
+    resourcesQuery += `${WorkflowConstants.ARCH_WORKFLOW_RESOURCE}/${workflowKey}`;
+
+    return ZosmfRestClient.deleteExpectString(session, resourcesQuery, []);
```
Comment (uzuko01) [inline packages/workflows/src/api/ArchivedDelete.ts]: maybe null instead of [] ?

File: packages/workflows/src/api/ArchivedDelete.ts (hunk 1 above)
Comment (melvl01) [reply packages/workflows/src/api/ArchivedDelete.ts]: done

File: packages/workflows/src/api/ArchivedDelete.ts (hunk 2 above)
Comment (melvl01) [reply packages/workflows/src/api/ArchivedDelete.ts]: ok

File: packages/workflows/src/api/ArchivedDelete.ts (hunk 3 above)
Comment (melvl01) [reply packages/workflows/src/api/ArchivedDelete.ts]: changed

File: packages/workflows/src/api/ArchivedDelete.ts (hunk 4 above)
Comment (melvl01) [reply packages/workflows/src/api/ArchivedDelete.ts]: changed

**Comment by zFernand0** [comment]
I'm curious about the rationale of having the "action" after the potential CLI "object".
For example, `zowe console issue command` has a corresponding API class called `IssueCommand` in the [IssueCommand.ts](https://github.com/zowe/zowe-cli/blob/master/packages/zosconsole/src/api/IssueCommand.ts) file

@melvl01, Should this be changed?

**Comment by zFernand0** [comment]
Also, Should this be associated with the already existing Delete APIs of the Workflow package/group?
I don't think we need a separate class if there is already a Delete class containing said workflow APIs.

**Comment by melvl01** [comment]
Thanks zFernand0, Will discuss both comments with team...

**Comment by zFernand0** [comment]
> Thanks zFernand0, Will discuss both comments with team...
@melvl01, No need to change how it was done.
FYI, if we do change it to use that `Delete.workflow` or `Delete.arvhivedWorkflow` model, then we will definitely have another breaking change, and this feature may bring further complications when porting it to the `lts-incremental` supported branch

**Pull request merged**